### PR TITLE
Bug 1387520 Add MariaDB Section

### PR DESCRIPTION
**Pull request by bmcelvee**
https://bugzilla.redhat.com/show_bug.cgi?id=1387520

### CODE REVIEW

**Comment by bmcelvee** [comment]
@adellape @ahardin-rh Please peer review if you have time. Thanks! 😸

**Comment by bmcelvee** [comment]
Preview: http://file.rdu.redhat.com/~bmcelvee/06042017/BZ1387520/using_images/db_images/mariadb.html

**Comment by vikram-redhat** [comment]
@bfallonf @mburke5678 - PTAL.

File: using_images/db_images/mariadb.adoc [hunk 1]
```diff
@@ -0,0 +1,754 @@
...
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[mariadb-overview]]
+== Overview
+{product-title} provides a container image for running MariaDB. This image can
+provide database services based on username, password, and database name
+settings provided via configuration.
```
Comment (mburke5678) [inline using_images/db_images/mariadb.adoc]: Dopey suggesting using "via" with caution. s/'settings in a configuration file (??)

File: using_images/db_images/mariadb.adoc [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+----
+
+This will create a container named `mariadb_database` running MySQL with
+database `db` and user with credentials `user:pass`. Port 3306 will be exposed
+and mapped to the host. If you want your database to be persistent across
+container executions, also add a `-v /host/db/path:/var/lib/mysql/data` argument.
+This will be the MySQL data directory.
+
+If the database directory is not initialized, the entrypoint script will first
+run `mysql_install_db` and setup necessary database users and passwords. After the
```
Comment (bfallonf) [inline using_images/db_images/mariadb.adoc]: s/setup/set up the

File: using_images/db_images/mariadb.adoc [hunk 3]
```diff
@@ -0,0 +1,754 @@
...
+----
++------------+------------------+
+| database() | user()           |
++------------+------------------+
+| sampledb   | root@172.17.42.1 |
++------------+------------------+
+----
+====
+
+== Creating a Database Service from a Template
```
Comment (bfallonf) [inline using_images/db_images/mariadb.adoc]: Anchor?

File: using_images/db_images/mariadb.adoc [hunk 4]
```diff
@@ -0,0 +1,754 @@
...
+be used in production.
+====
+endif::[]
+
+ifdef::openshift-enterprise[]
+[NOTE]
+====
+Enabling clustering for database images is currently in Technology Preview and
+not intended for production use.
+====
```
Comment (bfallonf) [inline using_images/db_images/mariadb.adoc]: I believe there's a template we're supposed to be using for tech previews now:

https://mojo.redhat.com/docs/DOC-1124368

Some of us chatted about it in #3952 , so not sure how much energy wants to go into it. Up to you.

File: using_images/db_images/mariadb.adoc [hunk 5]
```diff
@@ -0,0 +1,754 @@
...
+Since we claimed a persistent volume in this deployment configuration to have
+all data persisted for the MySQL master server, you must ask your cluster
+administrator to create a persistent volume that you can claim the storage from.
+
+After the deployment configuration is created and the pod with MariaDB master
+server is started, it will create the database defined by `MYSQL_DATABASE` and
+configure the server to replicate this database to slaves.
+
+The example provided defines only one replica of the MariaDB master server. This
+causes {product-title}to start only one instance of the server. Multiple
```
Comment (bfallonf) [inline using_images/db_images/mariadb.adoc]: Is there a missing space after the {product-title} thing above?

**Comment by bfallonf** [comment]
A few nits from me. Overall, this is fine, good job, etc :smile:  :truck:

File: using_images/db_images/mariadb.adoc [hunk 6]
```diff
@@ -0,0 +1,755 @@
...
+endif::[]
+
+There are two templates available:
+
+* `mysql-ephemeral` is for development or testing purposes only because it uses
+ephemeral storage for the database content. This means that if the database
+pod is restarted for any reason, such as the pod being moved to another node
+or the deployment configuration being updated and triggering a redeploy, all
+data will be lost.
+* `mysql-persistent` uses a persistent volume store for the database data
```
Comment (wzheng1) [inline using_images/db_images/mariadb.adoc]: Should be mariadb-persistent template, not mysql-persistent.

File: using_images/db_images/mariadb.adoc [hunk 7]
```diff
@@ -0,0 +1,755 @@
...
+The MariaDB templates should have been registered in the default *openshift*
+project by your cluster administrator during the initial cluster setup.
+ifdef::openshift-enterprise,openshift-origin[]
+See xref:../../install_config/imagestreams_templates.adoc#install-config-imagestreams-templates[Loading the Default Image Streams and Templates]
+for more details, if required.
+endif::[]
+
+There are two templates available:
+
+* `mysql-ephemeral` is for development or testing purposes only because it uses
```
Comment (wzheng1) [inline using_images/db_images/mariadb.adoc]: hould be mariadb-ephemeral template, not mysql-ephemeral.

**Comment by bmcelvee** [comment]
[rev_history]
|xref:../using_images/db_images/mariadb.adoc#using-images-db-images-mariadb[MariaDB]
|Added MariaDB section.
%

**Pull request merged**